shed/event_streams.py

```diff
@@ -231,37 +232,57 @@ def update(self, x, who=None):
         """
         return self.emit(self.dispatch(x))
 
-    def curate_streams(self, nds):
-        """Standardize name document pairs
+    def curate_streams(self, nds, outbound):
+        """Standardize inbound/outbound name document pairs
 
         Parameters
         ----------
         nds: tuple, or tuple of tuples
             The name document pair(s)
+        outbound: bool
+            If True curate streams outbound
 
         Returns
         -------
         name: str
             The name of the output doc(s)
-        docs: tuple
+        docs: {tuple, dict}
             The document(s)
 
         Notes
         ------
         If we get multiple streams make (name, (doc, doc, doc, ...))
         Otherwise (name, (doc,))
         """
-
-        if isinstance(nds[0], tuple):
-            names, docs = list(zzip(*nds))
-            if len(set(names)) > 1:
-                raise RuntimeError('Misaligned Streams')
-            name = names[0]
+        if not outbound:
+            # if there are multiple streams
+            if isinstance(nds[0], tuple):
+                names, docs = list(zzip(*nds))
+                if len(set(names)) > 1:
+                    raise RuntimeError('Misaligned Streams')
+                name = names[0]
+                newdocs = list()
+                for doc in docs:
+                    # for case of ((name, ({}, {})), (name, ({}, {})))
+                    if isinstance(doc, tuple):
+                        newdocs.extend(doc)
+                    else:
+                        newdocs.append(doc)
+
+                docs = tuple(newdocs)
+
+            # if only one stream
+            else:
+                names, docs = nds
+                name = names
+                docs = (docs,)
```

Comment:
we should check that docs is not a tuple here. if it is, then we shouldn't do this line
`docs = (docs,)`